spec/features/course/achievement_management_spec.rb

```diff
@@ -0,0 +1,164 @@
+require 'rails_helper'
+
+describe 'Achievement management' do
+  subject { page }
+
+  let!(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+    let!(:user) { create(:user, role: :administrator) }
+    let!(:course) { create(:course) }
+
+    before do
+      login_as(user, scope: :user)
+    end
+
+    describe 'achievement creation' do
+      before { visit new_course_achievement_path(course) }
+
+      context 'with invalid information' do
+        before { click_button 'Create' }
+
+        it 'stays on the same page' do
+          expect(page).to have_button('Create')
+        end
+
+        it 'shows errors' do
+          expect(page).to have_css('div.has-error')
+        end
+      end
+
+      context 'with valid information' do
+        let(:achievement) { build(:course_achievement, course: course) }
+
+        before do
+          fill_in 'course_achievement_title',    with: achievement.title
+          fill_in 'course_achievement_description',  with: achievement.description
+        end
+
+        it 'creates an achievement' do
+          expect { click_button 'Create' }.to change(Course::Achievement, :count).by(1)
+        end
+
+        context 'after creation' do
+          before { click_button 'Create' }
+
+          it 'shows the success message' do
+            expect(page).to have_selector('div',
+                                          text: I18n.translate('course.achievements.create.notice',
+                                                               title: achievement.title))
+          end
+
+          it 'redirects the user to the index page' do
+            expect(current_path).to eq(course_achievements_path(course))
+          end
+        end
+      end
+    end
+
+    describe 'achievement editing' do
+      let!(:achievement) { create(:course_achievement, course: course) }
+
+      before { visit edit_course_achievement_path(course, achievement) }
+
+      context 'page rendering' do
+        it { is_expected.to have_field('course_achievement_title', with: achievement.title) }
+        it { is_expected.to have_field('course_achievement_description',
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.